php/ext/google/protobuf/encode_decode.c

```diff
@@ -371,6 +369,21 @@ static void* str_handler(void *closure,
 }
 
 static bool str_end_handler(void *closure, const void *hd) {
+  stringfields_parseframe_t* frame = closure;
```

Comment:
Is this safe?

Reply:
Yes. str_end_handler is always paired with str_handler.
You can check add_handlers_for_singular_field where we register handlers for different fields.